### PR TITLE
Add BIP65 CHECKLOCKTIMEVERIFY to release notes

### PR DESCRIPTION
**Pull request by petertodd**
@luke-jr Would you please advise on what the release notes should say re: libblkmaker?

### CODE REVIEW

File: doc/release-notes.md [hunk 1]
```diff
@@ -124,6 +124,33 @@ sanity check. Since 0.12, these are no longer stored. When loading a
...
+Bitcoin Core's block templates are now for version 4 blocks only, and any
+mining software relying on its `getblocktemplate` must be updated in parallel
+to use either libblkmaker version FIXME or any version from 0.5.1 onward. If
+you are solo mining, this will affect you the moment you upgrade Bitcoin Core,
+which must be done prior to BIP65 achieving its 951/1001 status.  If you are
+mining with the stratum mining protocol: this does not affect you.  If you are
+mining with the getblocktemplate protocol to a pool: this will affect you at
+the pool operator's discretion, which must be no later than BIP65 achieving its
+951/1001 status.
+
```
Comment (btcdrak) [inline doc/release-notes.md]: Should we mention something about SPV clients here?

Comment (petertodd) [inline doc/release-notes.md]: Given the BIP mentions it, I don't think we need to here.

**Comment by jonasschnelli** [comment]
ACK

File: doc/release-notes.md [hunk 2]
```diff
@@ -124,6 +124,33 @@ sanity check. Since 0.12, these are no longer stored. When loading a
...
+additional restrictions to the NOP2 opcode. The same miner-voting mechanism as
+in BIP34 and BIP66 is used: when 751 out of a sequence of 1001 blocks have
+version number 4 or higher, the new consensus rule becomes active for those
+blocks. When 951 out of a sequence of 1001 blocks have version number 4 or
+higher, it becomes mandatory for all blocks and blocks with versions less than
+4 are rejected.
+
+Bitcoin Core's block templates are now for version 4 blocks only, and any
+mining software relying on its `getblocktemplate` must be updated in parallel
+to use either libblkmaker version FIXME or any version from 0.5.1 onward. If
```
Comment (jonasschnelli) [inline doc/release-notes.md]: Wouldn't it be sufficient to just use libblkmaker >= 0.5.1? Remove the "FIXME"?

**Comment by instagibbs** [comment]
ACK

File: doc/release-notes.md (hunk 2 above)
Comment (petertodd) [inline doc/release-notes.md]: Sure, but if Luke-Jr is going to release a backport, I'd like to know!

**Comment by petertodd** [comment]
@laanwj Um.... you realize this still have a FIXME in it, waiting for @luke-jr to tell me what version I should put in the release notes...

**Comment by laanwj** [comment]
That can be added in later (any time before the 0.12 release), I don't want to hold up a documentation pull for ages.

**Pull request merged**